### PR TITLE
Add persist id into field

### PR DESCRIPTION
**Pull request by yjiangcs**
@dheavern @bprang 

We added a new field `persistent_id` as UID for bulk provisioning. This field is an optional arg to call the api endpoint [members/add](https://www.dropbox.com/developers/documentation/http/teams#team-members-add).
This new field shall not affect the previous features.

### CODE REVIEW

File: Source/DfBAdminToolkit.Common/Services/MemberServices.cs [hunk 1]
```diff
@@ -145,6 +145,10 @@ public IServiceResponse AddMember(IMemberData data, string authToken)
                         ),
                         new JProperty("force_async", false)
                     );
+                    if (data.PersistentId != null) {
+                        JObject prov = (JObject)(jsonProv["new_members"] as JArray).First;
+                        prov.Property("member_surname").AddAfterSelf(new JProperty("member_persistent_id", data.PersistentId.Trim()));
```
Comment (dheavern) [inline Source/DfBAdminToolkit.Common/Services/MemberServices.cs]: Does order actually matter here? It'd be cleaner if we can just do prov.Add

File: Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs [hunk 2]
```diff
@@ -99,6 +99,10 @@ public bool LoadInputFile(IProvisioningModel model, IMainPresenter presenter)
                                     LastName = reader.GetField<string>(2),
                                     IsChecked = true
                                 };
+                                string persistent_id;
+                                if (reader.TryGetField<string>(3, out persistent_id)) {
```
[the remark this answers is not in the document]
Comment (yjiangcs) [reply Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs]: done

File: Source/DfBAdminToolkit.Common/Services/MemberServices.cs (hunk 1 above)
Comment (yjiangcs) [reply Source/DfBAdminToolkit.Common/Services/MemberServices.cs]: done. Changed to prov.add

File: Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs [hunk 3]
```diff
@@ -204,16 +213,19 @@ private string ProvisionRoles(IProvisioningModel model, IMainPresenter presenter
             {
                 foreach (MemberListViewItemModel item in model.Members.Where(m => m.IsChecked).ToList())
```
[the remark this answers is not in the document]
Comment (yjiangcs) [reply Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs]: Discussed offline. Decided to not do it in this PR.

File: Source/DfBAdminToolkit.Common/Services/MemberServices.cs [hunk 4]
```diff
@@ -145,6 +145,10 @@ public IServiceResponse AddMember(IMemberData data, string authToken)
                         ),
                         new JProperty("force_async", false)
                     );
+                    if (data.PersistentId != null) {
```
Comment (dheavern) [inline Source/DfBAdminToolkit.Common/Services/MemberServices.cs]: We should probably change this to `!String.IsNullOrEmpty(data.PersistentId)` if the data table columns are editable.
```suggestion
                    if (!String.IsNullOrEmpty(data.PersistentId)) {
```

File: Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs [hunk 5]
```diff
@@ -231,6 +239,16 @@ private string ProvisionRoles(IProvisioningModel model, IMainPresenter presenter
...
                                     presenter.UpdateProgressInfo("Team is already full. The organization has no available licenses.");
                                 }
+                                if (response.Message.Contains("persistent_id_disabled"))
+                                {
+                                    item.ProvisionStatus = "Persistent ID Disabled";
+                                    presenter.UpdateProgressInfo("Persistent ID Disabled");
+                                }
+                                if (response.Message.Contains("duplicate_member_persistent_id"))
+                                {
+                                    item.ProvisionStatus = "Duplicate member persistebt ID";
```
Comment (bprang) [inline Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs]: spelling

File: Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs [hunk 6]
```diff
@@ -231,6 +239,16 @@ private string ProvisionRoles(IProvisioningModel model, IMainPresenter presenter
...
+                                    presenter.UpdateProgressInfo("Duplicate member persistebt ID");
```
Comment (bprang) [inline Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs]: spelling

File: Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (yjiangcs) [reply Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs]: done

File: Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (yjiangcs) [reply Source/DfBAdminToolkit/Presenter/ProvisioningPresenter.cs]: done

**Pull request merged**